lib/rules/base.js

```diff
@@ -48,57 +43,300 @@ module.exports = function(options) {
-          pluginContext._processConfigNode(node);
+      // TODO: deprecation warning
+      pluginContext._processLegacyConfigNode(node);
+    });
+
+    // Handle element-child config statements
+    astHandlers.MustacheCommentStatement.enter.after.push(function(commentNode) {
+      if (inElementComments) {
+        return;
+      }
+
+      var config = pluginContext._processInstructionNode(commentNode);
+      if (!config) {
+        return;
+      }
+
+      pluginContext._pushConfig(config.value);
+
+      // Pop the config once we exit the comment's siblings
+      var parentNode = _.last(elementStack);
+      astHandlers.ElementNode.keys.children.exit.before.unshift(function(elementNode) {
+        if (elementNode === parentNode) {
+          pluginContext._popConfig(config.value);
+          return false;
+        }
+      });
+    });
+
+    // Handle in-element config statements
+    astHandlers.ElementNode.enter.before.push(function(elementNode) {
+      elementNode.comments.forEach(function(commentNode) {
+        var config = pluginContext._processInstructionNode(commentNode);
+        if (!config) {
+          return;
         }
 
-        commentVisitorEnter.apply(this, arguments);
-      };
-    }
+        pluginContext._pushConfig(config.value);
+
+        if (config.tree) {
+          // Applies to descendants, so pop the config once we exit the
+          // comment's element
+          astHandlers.ElementNode.exit.after.unshift(function(node) {
+            if (node === elementNode) {
+              pluginContext._popConfig(config.value);
+              return false;
+            }
+          });
+        } else {
+          // Applies to only this element, so pop the config when we enter the
+          // element's children, re-push it when we exit the element's
+          // children, and then re-pop it when we exit the element entirely.
+          astHandlers.ElementNode.keys.children.enter.before.unshift(function(node) {
+            if (node === elementNode) {
+              pluginContext._popConfig(config.value);
+              return false;
+            }
+          });
+          astHandlers.ElementNode.keys.children.exit.after.push(function(node) {
+            if (node === elementNode) {
+              pluginContext._pushConfig(config.value);
+              return false;
+            }
+          });
+          astHandlers.ElementNode.exit.after.unshift(function(node) {
+            if (node === elementNode) {
+              pluginContext._popConfig(config.value);
+              return false;
+            }
+          });
+        }
+      });
+    });
 
     return visitors;
   };
 
-  BasePlugin.prototype.processVisitor = function(ruleVisitor) {
-    var pluginContext = this;
-    var visitor = {
-      enter: function(node) {
-        if (!pluginContext.isDisabled()) {
-          invokeVisitorEnter(ruleVisitor, [node], pluginContext);
+  BasePlugin.prototype.visitors = function() { };
```

Comment:
Complete aside (for a future issue/PR): we should probably throw if `this.visitors()` returns a non-object...